core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -285,16 +285,25 @@ private[spark] object TestUtils {
   // minimum python supported version changes.
   val minimumPythonSupportedVersion: String = "3.7.0"
 
+  def assumePythonVersionAvailable: Unit =
+    assume(isPythonVersionAvailable,
```

Comment:
assume() will skip the test if this check fails. Should we perhaps fail? otherwise one would find that Python 3.6 tests all pass?

Reply:
```
Spark runs on Java 8/11/17, Scala 2.12/2.13, Python 3.7+ and R 3.5+. Python 3.7 support is deprecated as of Spark 3.4.0.
```

Do we still promise that Python 3.0 ~ Python 3.6 can pass all tests? If not, whether it is better to skip the test and explicitly notify?

I have the impression that some cases cannot pass when using Python 3.6, let me check it again

Reply:
> I have the impression that some cases cannot pass when using Python 3.6, let me check it again

Sorry, I didn't have time to check this today, will test later

Reply:
Python 3.6 support isn't promised. If one runs with Python 3.6, what should we do - not test at all? fail? try anyway?
assume() will not even try to test, which seems wrong. require() would fail, which seems reasonable. It also seems reasonable to do nothing, and just try the test anyway. So if there is any change here, seems like it would be to fail explicitly, not skip the test. I could see just not doing anything, too

Reply:
The original logic is `assume(TestUtils.testCommandAvailable("python3"))`

So if I want to do something, it should be `assume(TestUtils.testCommandAvailable("python3"))` and `require( py version >= 3.7)`?

Reply:
That's what I'd say. Or do nothing. Just let tests try, regardless of python version

Reply:
better keep still than move, let's leave it as it is, I will close this one